test/functional/wallet_fundrawtransaction.py

```diff
@@ -315,6 +316,67 @@ def test_change_type(self):
+        rawtx = wallet.createrawtransaction([], outputs)
+        funded_tx = wallet.fundrawtransaction(hexstring=rawtx, options={'changeAddress': change_addr})
+
+        # Verify change output existence
+        change_pos = funded_tx['changepos']
+        assert change_pos != -1
+
+        # Verify outputs now; the wallet must have increased the existent change out value instead of creating a new one
+        dec_tx = wallet.decoderawtransaction(funded_tx['hex'])
+        assert_equal(wallet.gettransaction(dec_tx['vin'][0]['txid'])['amount'], Decimal(50))  # assert input value
+        assert_equal(len(dec_tx['vout']), 2)
+        for index in range(2):
+            out = dec_tx['vout'][index]
+            if index == change_pos:
+                # Change must have increased its value, change = input - outputs - fee.
+                assert_equal(out['value'], Decimal(50) - outputs[external_addr] - funded_tx['fee'])
+                assert_equal(out['scriptPubKey']['address'], change_addr)
+            else:
+                # Non-change output must remain the same
+                assert_equal(out['value'], outputs[external_addr])
+                assert_equal(out['scriptPubKey']['address'], external_addr)
+
+        # 2) Fund tx and verify that the second output going to the change address is not discarded by the wallet
+        #    changeless txs predilection. Even when the output is "change" under the wallet perspective, the user has
+        #    set it manually, so it must not be discarded.
+        outputs = {external_addr: Decimal("49.998"), change_addr: Decimal(3)}
+        rawtx = wallet.createrawtransaction([], outputs)
+        funded_tx = wallet.fundrawtransaction(hexstring=rawtx, options={'changeAddress': change_addr})
+
+        # Verify change output existence
+        change_pos = funded_tx['changepos']
+        assert change_pos != -1
+
+        # Verify outputs once more; the wallet must have increased the existent change out value instead of removing it
+        # in favor of a changeless solution
+        dec_tx = wallet.decoderawtransaction(funded_tx['hex'])
+        # assert inputs value
+        assert_equal(len(dec_tx['vin']), 2)
+        for input in dec_tx['vin']:
+            assert_equal(wallet.gettransaction(input['txid'])['amount'], Decimal(50))
+
+        assert_equal(len(dec_tx['vout']), 2)
+        for index in range(2):
+            out = dec_tx['vout'][index]
+            if index == change_pos:
+                # Change must have increased its value, change = input - outputs - fee.
+                assert_equal(out['value'], Decimal(100) - outputs[external_addr] - funded_tx['fee'])
+                assert_equal(out['scriptPubKey']['address'], change_addr)
+            else:
+                # Non-change output must remain the same
+                assert_equal(out['value'], outputs[external_addr])
+                assert_equal(out['scriptPubKey']['address'], external_addr)
+
```

Comment:
Perhaps you should add a test that checks how much fees you’ve paid for the change outputs. I suspect you might pay two change fees when increasing an existing change, unless that gets cleaned up in some fashion. A test that rules it out would be neat.

Reply:
Yeah, it gets cleaned at the end of the tx creation process (check https://github.com/bitcoin/bitcoin/pull/27601#discussion_r1247203254).

But new tests never hurt, so just added it inside e46c1276. Thanks for pushing me to do it.